### ISSUE TITLE
Mistyped urlName in QueryParamConfig

### ISSUE THREAD

**Issue by aub** [pbeshai/use-query-params#227]
In [here](https://github.com/pbeshai/use-query-params/blob/caec1370ec0b6ba73e4c333a38fc81588e2f9a15/packages/serialize-query-params/src/types.ts#L42) the value passed to `urlName` for a `QueryParamConfig` is limited to the types of things that can  be decoded for the param, but I think it should be allowed to be any string. With the current typing, this is allowed:

```
const categoryParam = createEnumParam<JourneyCategoryEnum>(Object.values(JourneyCategoryEnum));
categoryParam.urlName = undefined;
categoryParam.urlName = JourneyCategoryEnum.all;
```

But this is not allowed:

```
const categoryParam = createEnumParam<JourneyCategoryEnum>(Object.values(JourneyCategoryEnum));
categoryParam.urlName = 'CATEGORY';
```

Since we're just specifying what we want to call the param in the URL, shouldn't any string be allowed? I changed the line linked above to:

```
urlName?: String;
```

And everything now works as I expected.

**Comment by pbeshai** [comment]
oh wow yes you're right! thank you!

**Comment by pbeshai** [comment]
should be fixed in serialize-query-params 2.0.1

**Comment by aub** [comment]
Nice, thanks for the fix! I had been trying to figure out how to get this functionality myself and kept failing, so I'm really happy to see it supported directly.
